tests/tests_data/conftest.py

```diff
@@ -0,0 +1,10 @@
+import pytest
+import torch.distributed
+
+
+@pytest.fixture(autouse=True)
+def teardown_process_group():
```

Comment:
I would like to understand why this is needed suddenly. Will I see an issue if I run the test without thix fixture?

Reply:
Otherwise I end up with not terminated processes since torch.distributed wasn't shut down. This causes tests to be a lot slower. We also have it in fabric and PL :)

Reply:
Oh right, but maybe delete the file here and add it in the other PR?

Reply:
but this one uses ddp as well :) so it should be added with whichever PR lands first :)

Reply:
This PR only makes fabric changes. I don't see how it relates to test_data. tests for data shouldn't even run on this PR, right?

Reply:
oh, yeah you're right, my bad